pkg/sql/parser/sql.y

```diff
@@ -16057,11 +16094,13 @@ unreserved_keyword:
 | SEQUENCE
 | SEQUENCES
 | SERVER
+| SERVICE
 | SESSION
 | SESSIONS
 | SET
 | SETS
 | SHARE
+| SHARED
```

Comment:
I've recently learned that we should potentially also add these new keywords to the bare_keywords list.  But, I plan to do a pass for all the keywords added in 23.1 soon.